scripts/checkProjectReferences.mjs

```diff
@@ -0,0 +1,65 @@
+import fs from "fs";
+import glob from "glob";
+import JSONC from "jsonc-parser";
+import path from "path";
+import url from "url";
+
+
+const __filename = url.fileURLToPath(new URL(import.meta.url));
+const __dirname = path.dirname(__filename);
+
+
+// This script checks that we list all transitive references in all tsconfig.json files.
+// See: https://github.com/microsoft/TypeScript/issues/30608
+
+
+/**
+ * @param {string} p
+ */
+function getTsconfigPath(p) {

```

Comment:
I don't think the glob can produce either special case; if that's true, then I'd remove this function

Reply:
You need this because you have to traverse `references.path` inside of tsconfig, and we don't put `.json` extensions in those (though they are legal).